woocommerce/woocommerce
# Add description and delivery time to shipping methods in the Cart and Checkout blocks

### Changes proposed in this Pull Request:

Adds two new properties to the data of a `WC_Shipping_Rate`: `delivery_time` and `description`.

Both are strings, and they will be used to display a delivery time, and a description of the shipping rate in the Checkout block.

The design is here, however we are not going to implement the lighter colour of the delivery time due to issues with different coloured themes.

![image](https://github.com/user-attachments/assets/45b37b24-62f4-4cf7-acaf-0ca4cadc1d5b)

The code for displaying these properties already exists in the Checkout block, but it was never used as these properties didn't exist.





Closes #42667



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Add several shipping methods to your store. A mix of free and flate rate shipping options should be used. Local pickup _can_ be added, but isn't relevant to this, as it depends on the extension's use of `delivery_time` and `description` props, rather than anything we control.
2. Add the following code using a plugin like [Code Snippets](https://wordpress.org/plugins/code-snippets/)
```php
function custom_shipping_description( $rates, $package ) {
    foreach ( $rates as $rate_id => $rate ) {
		if ( 'pickup_location' === $rate->method_id ) {
			continue;
		}
		$rate->description = 'This is a custom description for ' . $rate->label . '.';
		$rate->delivery_time = 'This will be delivered in ' . rand( 1, 5 ) . ' days.';
	}
    return $rates;
}
add_filter( 'woocommerce_package_rates', 'custom_shipping_description', 10, 2 );
```
4. Ensure you have `WooCommerce -> Settings -> Shipping -> Shipping settings -> Enable the shipping calculator on the cart page` checked.
5. Add an item to your cart and go to the Cart block.
6. Ensure you see the rate descriptions and delivery times in the shipping calculator, like so:
<img width="343" alt="image" src="https://github.com/user-attachments/assets/36860a09-54db-40c9-8677-0dc0375a4efa">

7. Go to the Checkout block. Ensure you see the descriptions and delivery times in the shipping methods block, like so:
<img width="576" alt="image" src="https://github.com/user-attachments/assets/c23a5500-1888-4bc3-863e-bda8a662c913" />

8. Change the text in the snippet above to be quite long, e.g. like so:
```php
function custom_shipping_description( $rates, $package ) {
    foreach ( $rates as $rate_id => $rate ) {
		if ( 'pickup_location' === $rate->method_id ) {
			continue;
		}
		$rate->description = 'This is a custom description for ' . $rate->label . '. It is quite a long, descriptive description that should flow onto multiple lines, for testing purposes.';
		$rate->delivery_time = 'This will be delivered in ' . rand( 1, 5 ) . ' days.';
	}
    return $rates;
}
add_filter( 'woocommerce_package_rates', 'custom_shipping_description', 10, 2 );
```
9. Refresh the Checkout block and ensure it looks OK with the text on multiple lines. There should be no widows/runts (single words on their own line at the end of a block of text) in the description or delivery time.

<img width="573" alt="image" src="https://github.com/user-attachments/assets/a78637a4-cb4a-454b-ab8b-d5a9ff031f35" />

10. Try with different screen sizes, and on mobile and various different browsers.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
